Add release flag (#1227)
hyperledger/solang: docs/code_gen_options.rst

Reviewer: The `--release` flag should be mentioned in the CLI guide: https://solang.readthedocs.io/en/latest/running.html
In addition, this information should also be clear in the example.

Reviewer 2: @salaheldinsoliman `--release` is not mentioned in `docs/running.rst`

Author: I mentioned it, thanks

Debugging Options
-----------------

It is desirable to have access to debug information regarding the contract execution in the testing phase.
Hence, debugging options are enabled by default, but can be disbaled using CLI flags. 
Therefore, by default, debugging options are enabled; however, they can be deactivated by utilizing the command-line interface (CLI) flags.

> It is imperative to disable debugging options in release builds, as debugging would result in increased instructions and greater gas consumption.

Reviewer 2: `imperative` has a very strong connotation, which is totally uncalled for here.
```suggestion
Debugging options are disabled in release builds, as debug builds greatly increase contract size and gas consumption.
```

> Solang provides three debugging options, namely debug prints, logging API return codes, and logging runtime errors.

Reviewer 3: I think this sentence should be moved to the beginning of this chapter (`Debugging Options`). It should also say that these 3 options are exactly the ones being disable by the `--release` flag.

> .. _no-print:

Reviewer: You have included markers on top of every section you wrote here, but you have not used them anywhere. Please, only create markers if you are going to use them.

Author: Thanks, I will use them in the CLI guide

Print Function

> ++++++++++++++

Reviewer: I think this should be called `Debugging Logs`, not `Print Function`. Print function is an internal builtin and your CLI option does not enable or disable it.

Reviewer 2: @LucasSte it does disable `print()`

> Solang provides a ``print()`` functionality mainly for debugging. This feature  is enabled by default.

Reviewer 3: ```suggestion
Solang provides a `print builtin <https://solang.readthedocs.io/en/latest/language/builtins.html#print-string>` which  is enabled by default.
```

> Solang provides a `print builtin <https://solang.readthedocs.io/en/latest/language/builtins.html#print-string>` which is enabled by default.

Reviewer: For references inside the same document, we do not use links. We use cross-references, like these ones:

https://github.com/hyperledger/solang/blob/242c187fa981b523102c0147f4ed272836c01f69/docs/language/managing_values.rst?plain=1#L46

https://github.com/hyperledger/solang/blob/242c187fa981b523102c0147f4ed272836c01f69/docs/language/types.rst?plain=1#L163

Reviewer: This is important to maintain coherence if we someday change the website domain or the organization of the docs. A cross-reference will never brake.

> The ``no-log-prints`` flag will instruct the compiler not to log debugging prints in the environment.

Reviewer 3: ```suggestion
The ``no-print`` flag will instruct the compiler to omit compiling any print statement into the contract artifact.
```

.. _no-log-api-return-codes:

Log runtime API call results
++++++++++++++++++++++++++++

Runtime API calls are not guaranteed to succeed.
By design, the low level results of these calls are abstracted away in Solidity.
For development purposes, it can be desirable to observe the low level return code of such calls.

> The contract will print the return code of runtime calls by default, and this feature could be disabled by providing the 

Reviewer: ```suggestion
The contract will print the return code of runtime calls by default, and this feature can be disabled by providing the 
```

``--no-log-api-return-codes`` flag.

.. note::

    This is only implemented for the Substrate target.

.. _no-log-runtime-errors:

Log Runtime Errors
++++++++++++++++++

In most cases, contract execution will emit a human readable error message in case a runtime error is encountered.
The error is printed out alongside with the filename and line number that caused the error.

> This feature is enabled by default, and could be disabled by the ``--no-log-runtime-errors`` flag.

Reviewer: ```suggestion
This feature is enabled by default, and can be disabled by the ``--no-log-runtime-errors`` flag.
```

.. _release:

Release builds:
+++++++++++++++

Release builds must not contain any debugging related logic. The ``--release`` flag will turn off all debugging features, 

> therefore reducing the required gas and storage.

Reviewer: ```suggestion
thereby reducing the required gas and storage.
```
